src/components/access-type.js

```diff
@@ -321,23 +335,26 @@ class AccessTypeBase extends React.Component {
           storySlug,
         };
   }
-  //TODO -> need to write test cases to cover all scenarios , selectedPlan, planType , coupon, urls, story details etc.
+
   initRazorPayPayment = async (
     selectedPlanObj = {},
     planType = "",
     storyId = "",
     storyHeadline = "",
     storySlug = "",
-    paymentType = ""
+    paymentType = "",
+    opts
   ) => {
     if (!selectedPlanObj) {
       console.warn("Razor pay needs a plan");
       return false;
     }
-
+    const intent = get(opts, ["intent"], "default");
+    const switchType = get(opts, ["switchType"]);
     const planObject = this.makePlanObject(selectedPlanObj, planType, storyId, storyHeadline, storySlug); //we are doing this to sake of backward compatibility and will be refactored later.
     let { paymentOptions } = this.props;
     const { discounted_price_cents, price_cents } = planObject.selectedPlan;
+    const paymentObject = this.makePaymentObject({ ...planObject, couponCode: selectedPlanObj.coupon_code });
```

Comment:
shouldn't the **paymentObject** be declared after **planObject["paymentType"]** is assigned the value(line no: 364) bcz we are using the 
planObject

Reply:
yes good catch, made change